libraries/ESP8266WiFi/examples/WiFiTelnetToSerial/WiFiTelnetToSerial.ino

```diff
@@ -20,83 +20,190 @@
+#endif
+  logger->println(ESP.getFullVersion());
+  logger->printf("Serial baud: %d (8n1: %d KB/s)\n", BAUD_SERIAL, BAUD_SERIAL * 8 / 10 / 1024);
+  logger->printf("Serial receive buffer size: %d bytes\n", RXBUFFERSIZE);
+
+#if SERIAL_LOOPBACK
+  USC0(0) |= (1 << UCLBE); // incomplete HardwareSerial API
+  logger->println("Serial Internal Loopback enabled");
+#endif
+
   WiFi.mode(WIFI_STA);
   WiFi.begin(ssid, password);
-  Serial1.print("\nConnecting to "); Serial1.println(ssid);
-  uint8_t i = 0;
-  while (WiFi.status() != WL_CONNECTED && i++ < 20) {
+  logger->print("\nConnecting to ");
+  logger->println(ssid);
+  while (WiFi.status() != WL_CONNECTED) {
+    logger->print('.');
     delay(500);
   }
-  if (i == 21) {
-    Serial1.print("Could not connect to"); Serial1.println(ssid);
-    while (1) {
-      delay(500);
-    }
-  }
-  //start UART and the server
-  Serial.begin(115200);
+  logger->println();
+  logger->print("connected, address=");
+  logger->println(WiFi.localIP());
+
+  //start server
   server.begin();
   server.setNoDelay(true);
 
-  Serial1.print("Ready! Use 'telnet ");
-  Serial1.print(WiFi.localIP());
-  Serial1.println(" 23' to connect");
+  logger->print("Ready! Use 'telnet ");
+  logger->print(WiFi.localIP());
+  logger->printf(" %d' to connect\n", port);
 }
 
 void loop() {
-  uint8_t i;
   //check if there are any new clients
   if (server.hasClient()) {
-    for (i = 0; i < MAX_SRV_CLIENTS; i++) {
-      //find free/disconnected spot
-      if (!serverClients[i] || !serverClients[i].connected()) {
-        if (serverClients[i]) {
-          serverClients[i].stop();
-        }
+    //find free/disconnected spot
+    int i;
+    for (i = 0; i < MAX_SRV_CLIENTS; i++)
+      if (!serverClients[i]) { // equivalent to !serverClients[i].connected()
         serverClients[i] = server.available();
-        Serial1.print("New client: "); Serial1.print(i);
+        logger->print("New client: index ");
+        logger->print(i);
         break;
       }
-    }
+
     //no free/disconnected spot so reject
     if (i == MAX_SRV_CLIENTS) {
-      WiFiClient serverClient = server.available();
-      serverClient.stop();
-      Serial1.println("Connection rejected ");
+      server.available().println("busy");
+      // hints: server.available() is a WiFiClient with short-term scope
+      // when out of scope, a WiFiClient will
+      // - flush() - all data will be sent
+      // - stop() - automatically too
+      logger->printf("server is busy with %d active connections\n", MAX_SRV_CLIENTS);
     }
   }
-  //check clients for data
-  for (i = 0; i < MAX_SRV_CLIENTS; i++) {
-    if (serverClients[i] && serverClients[i].connected()) {
-      if (serverClients[i].available()) {
-        //get data from the telnet client and push it to the UART
-        while (serverClients[i].available()) {
-          Serial.write(serverClients[i].read());
+
+  //check TCP clients for data
+#if 1
```

Comment:
what's this?

Reply:
it's temporary, see the comment below
// Incredibly, this code is faster than the bufferred one below - #4620 is needed